src/parser/GenericParser.ts

```diff
@@ -38,40 +45,67 @@ export class GenericParser {
         let couplingMetrics = {} as CouplingResult;
 
         try {
-            const parseFilesGenerator = findFilesAsync(this.config);
+            const filePathGenerator = findFilesAsync(this.config);
+            const parsePromises = new Map<string, Promise<ParsedFile | SimpleFile>>();
+
+            for await (const file of filePathGenerator) {
+                const fileExtensionLanguage = fileExtensionToLanguage.get(file.fileExtension);
+                if (fileExtensionLanguage !== undefined) {
+                    parsePromises.set(
+                        file.filePath,
+                        TreeParser.parse(file, fileExtensionLanguage).catch((reason) => {
+                            console.error(
+                                "Error while parsing a tree for the file " + file.filePath
+                            );
+                            console.error(reason);
```

Comment:
This is indead something we should display to the users, so that they are aware that some files could not be parsed.